src/backend/executor/bridge.ts

```diff
@@ -0,0 +1,402 @@
+                this._bridgeMessages.fire(`>>> Database found at path: ${filePath}\n`);
+                // TODO: Cache the result and only update on eg. watch change
+                return filePath;
+            }
+        }
+
+        this._bridgeMessages.fire('>>> No database found in the following paths:\n');
+        for (const filePath of this.databasePaths) {
+            if (filePath) {
+                this._bridgeMessages.fire(`>>>   ${filePath}\n`);
+            } else {
+                this._bridgeMessages.fire('>>>   <no path set in settings>\n');
+            }
+        }
+
+        return undefined;
+    }
+
+    public getAnalyzeCmdLine(...files: Uri[]): string | undefined {
+        if (!workspace.workspaceFolders?.length) {
+            return undefined;
+        }
+
+        // TODO: Refactor for less code repetition across functions
+        const workspaceFolder = workspace.workspaceFolders[0].uri.fsPath;
+
+        const ccPath = getConfigAndReplaceVariables('codechecker.executor', 'executablePath')
+            ?? 'CodeChecker';
+        const ccFolder = getConfigAndReplaceVariables('codechecker.backend', 'outputFolder')
+            ?? path.join(workspaceFolder, '.codechecker');
+        const ccArguments = getConfigAndReplaceVariables('codechecker.executor', 'arguments') ?? '';
+        const ccThreads = workspace.getConfiguration('codechecker.executor').get<string>('threadCount');
+
+        const ccCompileCmd = this.getCompileCommandsPath();
+
+        if (ccCompileCmd === undefined) {
+            window.showWarningMessage('No compilation database found, CodeChecker not started - see logs for details');
+            return undefined;
+        }
+
+        const filePaths = files.length
+            ? `--file ${files.map((uri) => `"${uri.fsPath}"`).join(' ')}`
+            : '';
+
+        return [
+            `${ccPath} analyze`,
+            `"${ccCompileCmd}"`,
+            `--output "${ccFolder}"`,
+            `${ccThreads ? '-j ' + ccThreads : ''}`,
+            `${ccArguments}`,
+            `${filePaths}`,
+        ].join(' ');
+    }
+
+    public getLogCmdLine(): string | undefined {
+        if (!workspace.workspaceFolders?.length) {
+            return undefined;
+        }
+
+        const workspaceFolder = workspace.workspaceFolders[0].uri.fsPath;
+
+        const ccPath = getConfigAndReplaceVariables('codechecker.executor', 'executablePath')
+            ?? 'CodeChecker';
+        const ccFolder = getConfigAndReplaceVariables('codechecker.backend', 'outputFolder')
+            ?? path.join(workspaceFolder, '.codechecker');
+
+        // Use a predefined path here
+        const ccCompileCmd = path.join(ccFolder, 'compile_commands.json');
+
+        return [
+            `${ccPath} log`,
+            `--output "${ccCompileCmd}"`,
+            '--build "make"'
+        ].join(' ');
+    }
+
+    public getParseCmdLine(...files: Uri[]): string | undefined {
```

Comment:
We run the parse command on all 'visible text editors changed' events (opens, tab switches, etc.):
https://github.com/Ericsson/CodecheckerVSCodePlugin/blob/2d14270d45af306324a4b235399ecd214acfe946/src/backend/processor/diagnostics.ts#L60

I'm not sure how it handles opening new windows, since it's a VS Code builtin, but it hasn't changed from before.  
Will need to investigate further about whether this is an issue or not, but I don't believe a fix will make it into this PR.